crates/red_knot_python_semantic/src/types.rs

```diff
@@ -325,6 +325,8 @@ pub enum Type<'db> {
     ModuleLiteral(File),
     /// A specific class object
     ClassLiteral(ClassLiteralType<'db>),
+    // The set of all class objects that are subclasses of the given class (C), spelled `type[C]`.
+    Type(ClassLiteralType<'db>),
```

Comment:
Can we create a `TypeType` struct, that wraps a `Class`, rather than reusing the `ClassLiteralType` struct for this variant? Conceptually `TypeType` and `ClassLiteralType` are quite distinct things (though related, of course).

I guess the reason not to do this would be having to unpack it as `Type::Type(TypeType { class})` 😆 yet another reason to call it something else ;)

Reply:
I agree that we probably shouldn't use `ClassLiteralType` here. We could have it wrap `Class` directly, but if we want to stay consistent with using wrapper structs, even when they have only one element, then I agree we should create a new one, and it should be named `XType`, where `X` is the name we choose for the `Type::` variant itself. (So for example, `Type::SubclassOf(SubclassOfType)`.)

Reply:
Done.